### PR TITLE
Make `lu` results have same storage mode as input

### PR DESCRIPTION
**Pull request by christiangnrd**
Noticed while reviewing #313.

### CODE REVIEW

File: test/mps/linalg.jl [hunk 1]
```diff
@@ -191,22 +191,29 @@ end
 end
 
 @testset "decompositions" begin
+    testreturntype(_,_) = false
+    testreturntype(luobj::LU{<:Any,<:MtlArray{<:Any,<:Any,S},<:MtlArray{<:Any,<:Any,S}},::MtlArray{<:Any,<:Any,S}) where S = true
+
```
Comment (maleadt) [inline test/mps/linalg.jl:196]: Is it possible to do something like `@test storagemode(parent(lua)) == storagemode(A)`, or similar? That seems more self-documenting to me.

Comment (christiangnrd) [reply test/mps/linalg.jl]: Fixed

**Review by tgymnich — APPROVED** [review]

File: lib/mps/linalg.jl [hunk 2]
```diff
@@ -176,7 +176,7 @@ end
     dev = device()
     queue = global_queue(dev)
 
-    At = MtlMatrix{T,PrivateStorage}(undef, (N, M))
```
Comment (tgymnich) [inline lib/mps/linalg.jl:179]: `At` is only used by us and therefore should use the fastest possible StorageMode (private)

**Comment by christiangnrd** [comment]
Turns out the code was wrong, but so were the tests. Should be fixed now.

**Pull request merged**